i3lock-color: fix manpage-name and manpage

rkoe opened the pull request:
###### Motivation for this change
The manpage of i3lock-color should also be named i3lock-color (and not i3lock), and talk about i3lock-color (and not i3lock), so:

- rename manpage from i3lock.1 to i3lock-color.1
- change "i3lock" to "i3lock-color" in manpage-synopsis

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @garbas @malyn

rkoe committed:
i3lock-color: fix manpage-name and manpage

- rename manpage from i3lock.1 to i3lock-color.1
- change "i3lock" to "i3lock-color" in manpage-synopsis

xeji commented on pkgs/applications/window-managers/i3/lock-color.nix:
```diff
@@ -25,6 +25,8 @@ stdenv.mkDerivation rec {
+    sed -i 's/^\.B i3lock$/.B i3lock-color/' $out/share/man/man1/i3lock-color.1
   '';
   meta = with stdenv.lib; {
     description = "A simple screen locker like slock";
```
While you're at it, can you please add a hint here on how this differs from regular `i3lock`, for example
"A simple screen locker like slock, enhanced version with extra color options" 
(feel free to come up with a better wording).

rkoe replied:
Ok. This will be in a separate pull-request.

rkoe committed:
i3lock-color: fix manpage and rename pam.d-file

complete name-change from i3lock to i3lock-color:

- rename /etc/pam.d/i3lock to /etc/pam.d/i3lock-color
- change "i3lock" to "i3lock-color" everywhere in the manpage

rkoe commented:
Now, I've completely renamed everything:
- /etc/pam.d/i3lock to i3lock-color
- every occurence of "i3lock" to "i3lock-color" in the manpage.

rkoe commented on pkgs/applications/window-managers/i3/lock-color.nix:
```diff
@@ -25,6 +25,8 @@ stdenv.mkDerivation rec {
   postInstall = ''
     mv $out/bin/i3lock $out/bin/i3lock-color
+    mv $out/share/man/man1/i3lock.1 $out/share/man/man1/i3lock-color.1
+    sed -i 's/^\.B i3lock$/.B i3lock-color/' $out/share/man/man1/i3lock-color.1
```
Ok, I've extended it.

xeji commented on pkgs/applications/window-managers/i3/lock-color.nix:
```diff
@@ -25,6 +25,9 @@ stdenv.mkDerivation rec {
   installFlags = "PREFIX=\${out} SYSCONFDIR=\${out}/etc MANDIR=\${out}/share/man";
   postInstall = ''
     mv $out/bin/i3lock $out/bin/i3lock-color
+    mv $out/etc/pam.d/i3lock $out/etc/pam.d/i3lock-color
```
Please remove this line as it will cause problems with PAM authentication. The app starts PAM under the service name 'i3lock', so  the file must be named `/etc/pam.d/i3lock`.

xeji reviewed (changes requested):
Manpage looks good now but /etc/pam.d/i3lock should not be renamed.

rkoe committed:
i3lock-color: revert renaming of /etc/pam.d/i3lock

xeji reviewed (approved):
Thank you, LGTM now.

xeji commented:
@GrahamcOfBorg  build i3lock-color

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40095&attempt_id=bfd0d56d-dfce-4c9d-a19d-906627cf3d09)

The following builds were skipped because they don't evaluate on aarch64-linux: i3lock-color

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40095&attempt_id=0b8f3cd5-c238-4f54-940c-df9e0e6af3f3)

Attempted: i3lock-color

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c
shrinking /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c/bin/i3lock-color
gzipping man pages under /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c/share/man/
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c/bin
patching script interpreter paths in /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c
checking for references to /build in /nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c...
/nix/store/bcwc9z9c9pva3h5y7ldxlm2icid82xy2-i3lock-color-2.11-c
```
</p></details>
